components/widgets/menu/menu_items/submenu_item_action.jsx

```diff
@@ -0,0 +1,140 @@
+// Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
+// See LICENSE.txt for license information.
+
+import React from 'react';
+import PropTypes from 'prop-types';
+
+import * as Utils from 'utils/utils.jsx';
+import {ModalIdentifiers} from 'utils/constants';
+import SubMenuModal from '../../../submenu_modal/submenu_modal.jsx';
+
+export default class SubMenuItemAction extends React.PureComponent {
+    static propTypes= {
+        id: PropTypes.string,
+        text: PropTypes.oneOfType([PropTypes.string, PropTypes.node]).isRequired,
+        subMenu: PropTypes.arrayOf(PropTypes.object),
+        icon: PropTypes.node,
+        action: PropTypes.func.isRequired,
+        openModal: PropTypes.func,
+        xOffset: PropTypes.number,
+        ariaLabel: PropTypes.string,
+        root: PropTypes.bool,
+    };
+    static defaultProps = {
+        show: true,
+    };
+
+    constructor(props) {
+        super(props);
+        this.node = React.createRef();
+
+        this.state = {
+            show: false,
+        };
+    }
+
+    show = () => {
+        this.setState({show: true});
+    }
+
+    hide = () => {
+        this.setState({show: false});
+    }
+
+    handleShowMobileSubMenuItem = (elements, action) => {
+        const openModal = this.props.openModal;
+        const submenuModalData = {
+            ModalId: ModalIdentifiers.PLUGIN_MOBILE_SUBMENU,
```

Comment:
Maybe this could be just `MOBILE_SUBMENU` instead of `PLUGIN_MOBILE_SUBMENU` so we don't marry it to plugin and the `SubmenuModal` component can later be re-used in the webApp if we add a fixed submenu at any point.

Reply:
Great idea.  Done.